cwl/src/main/scala/cwl/CwlDecoder.scala

```diff
@@ -54,9 +54,9 @@ object CwlDecoder {
     * Notice it gives you one instance of Cwl.  This has transformed all embedded files into scala object state
     */
   def decodeCwlFile(fileName: BFile,
-                    workflowRoot: Option[String] = None): Parse[Cwl] =
+                    workflowRoot: Option[String] = None)(implicit processor: CwlPreProcessor = cwlPreProcessor): Parse[Cwl] =
```

Comment:
ToL I'm not crazy about implicitly importing non-typeclassy stuff  but I think the alternative, as @mcovarr alluded to in a previous PR, is the `Reader` Monad, which would make `Parse` more complicated.

I think that's ultimately the way to go, but until then I think we do it this way until we have time to plow through uncharted territory.